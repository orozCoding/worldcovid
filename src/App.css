@import url('https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&display=swap');
@import url("https://cdn.jsdelivr.net/npm/bootstrap-icons@1.8.1/font/bootstrap-icons.css");

/* global classes */ 

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.lato {
  font-family: 'Lato', sans-serif;;
}

.bold {
  font-weight: bold;
}

.d-flex {
  display: flex;
  justify-content:  center;
  align-items: center;
}

.col {
  flex-direction: column;
}

.ck {
  cursor: pointer;
}

html {
  text-align: center;
  width: 100%;
  background-color: white;
}

body {
  max-width: 500px;
  background-color: grey;
  color: white;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
  overflow: scroll;
  text-align: center;
  margin:auto;
}

#root {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%
}

.main {
  width: 100%;
  background-color: #4369b2;
  min-height: 97vh;
}

/* end of global classes */ 

.home-left,
.home-right {
  min-width: 50%;
  padding: 50px 0;
  height: 150px;
}

.home-left {
  background-color: #3d62a7;
}

.mainTitle {
  font-size: 30px;
  font-weight: bold;
}

.subTitle {
  width: 100%;
  font-size: 20px;
}

.separator {
  padding: 10px 0;
  border-top: solid 2px #3b5ea0;
  border-bottom: solid 2px #3b5ea0;
}

.totalTitle {
  font-size: 18px;
}

.countriesContainer {
  flex-wrap: wrap;
}

.countryBox {
  width: 50%;
  height: 150px;
}

.countryBox:hover {
  background-color: #4870bb;
}

.countryBox:nth-child(4n-1),
.countryBox:nth-child(4n-2) {
  background-color: #3d62a7;
}

.countryBox:nth-child(4n-1):hover,
.countryBox:nth-child(4n-2):hover {
  background-color: #385996;
}

.arrowIcon {
  height: 20px;
  padding: 15px 10px;
  width: 100%;
  align-self: flex-end;
  justify-content: end;
}

.arrowIcon i {
  margin-left: auto;
}

.countryInfo {
  width: 100%;
  margin-top: -35px;
  height: 150px;
  text-align: right;
}

.countryInfo div {
  margin-left: auto;
  padding-right: 10px;
}

.countryName {
  font-size: 25px;
}

/* single country style */

.selfCountries {
  width: 100%;
  justify-content: start;
  align-items: flex-start;
  margin-bottom: auto;
}

.countryContainer {
  width: 100%;
  flex-grow: 1;
}

.selfTitle {
  width: 100%;
  font-size: 60px;
  min-height: 130px;
  justify-content: end;
  padding: 20px;
}


.selfDate {
  width: 100%;
  gap: 30px;
  background-color: #3b5ea0;
  padding: 10px 0;
  font-size: 20px;
}

.selfStats {
  width: 100%;
}

.selfStats div {
  font-size: 20px;
  width: 100%;
  padding: 10px 80px;
  justify-content: space-between;
}

.selfStats div:nth-child(even) {
  background-color: #3b5ea0;
}

.backButton {
  padding: 30px;
  font-size: 35px;
}