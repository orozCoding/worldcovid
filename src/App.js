import React, { useEffect } from 'react';
import './App.css';
import WorldHome from './components/world/worldHome';
import { BrowserRouter as Router, Routes, Route, Navigate } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { loadCountriesData } from './redux/countries/countries';
import { loadWorldData } from './redux/world/world';
import SingleCountry from './components/countries/singleCountry';
import filterName from './components/filterName';

const App = () => {
  const countriesData = useSelector((state) => state.countries);
  const worldData = useSelector((state) => state.world);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!countriesData.length) {
      dispatch(loadCountriesData());
    }
  }, []);

  useEffect(() => {
    if (!worldData.length) {
      dispatch(loadWorldData());
    }
  }, []);


  return (
    <Router>
      <main className="main d-flex col">
        <Routes>
          <Route path="/" element={<WorldHome worldData={worldData} countriesData={countriesData} />} />
          {
            countriesData.map((country) => {
              return <Route key={country.id} path={
                `/${filterName(country.name)}`
              } element={<SingleCountry key={country.id} country={country} />} />
            })
          }
          <Route
            path="*"
            element={<Navigate to="/" />}
          />
        </Routes>
      </main>
    </Router>
  );
}

export default App;
