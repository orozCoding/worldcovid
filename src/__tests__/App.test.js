import React from 'react';
import {
  render, screen, fireEvent, wait,
} from '@testing-library/react';
import renderer, { act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../redux/configStore';
import worldInfo from '../components/testsComponents/statsMock';

let mockData;

beforeEach(() => {
  jest.spyOn(global, 'fetch').mockResolvedValue({
    json: jest.fn().mockResolvedValue(mockData),
  });
});

afterEach(() => {
  jest.restoreAllMocks();
});

describe('App', () => {
  mockData = worldInfo;
  it('renders', () => {
    let tree;
    act(() => {
      tree = renderer
        .create(
          <Provider store={store}>
            <App />
          </Provider>,
        )
        .toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('renders wold info', async () => {
    act(() => {
      render(<Provider store={store}><App /></Provider>);
    });
    await expect(screen.getByText('COVID19 GLOBAL STATS')).toBeDefined();
  });

  it('Renders countries data', async () => {
    act(() => {
      render(<Provider store={store}><App /></Provider>);
    });
    const country = act(() => { screen.getByText('Afghanistan'); });
    await expect(country).toBeDefined();
  });

  it('visits a country', async () => {
    act(() => {
      render(<Provider store={store}><App /></Provider>);
    });
    act(() => {
      fireEvent.click(screen.getByText('Afghanistan'));
    });
    const cases = screen.getByText('Cases today:');
    await expect(cases).toBeDefined();
  });
});
