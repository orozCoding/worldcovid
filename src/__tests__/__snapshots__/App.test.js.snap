// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<main
  className="main d-flex col"
>
  <div>
    <div
      className="homeHeader d-flex"
    >
      <div
        className="home-left d-flex col"
      >
        <p
          className="covid-title mainTitle lato"
        >
          COVID19 GLOBAL STATS
        </p>
      </div>
      <div
        className="home-right d-flex col"
      >
        <div
          className="subTitle lato bold"
        >
          Worldwide Stats:
        </div>
        <div
          className="d-flex worldData"
        >
          <p>
            Date:
          </p>
          <p>
            2022-03-07
          </p>
        </div>
        <div
          className="d-flex worldData"
        >
          <p>
            Total cases:
          </p>
          <p>
            0
          </p>
        </div>
        <div
          className="d-flex worldData"
        >
          <p>
            Total deaths:
          </p>
          <p>
            0
          </p>
        </div>
      </div>
    </div>
    <div
      className="separator d-flex col"
    >
      <p
        className="lato bold totalTitle"
      >
        Total Cases By Country
      </p>
      <p>
        Select one country for more information
      </p>
    </div>
    <div
      className="countriesContainer d-flex"
    >
      <div
        aria-hidden="true"
        className="countryBox d-flex col ck"
        onClick={[Function]}
      >
        <p>
          Source: Johns Hopkins University.
        </p>
        <p>
          API by Narrativa
        </p>
      </div>
    </div>
  </div>
</main>
`;
